cf-revision-manager.php

```diff
@@ -3,199 +3,303 @@
-			add_action('wp_restore_post_revision', array($this, 'restore_post_revision'), 10, 2);
-
-			if (is_admin()) {		
-				# revision display
-				global $pagenow;
-				if ($pagenow == 'revision.php') {
-					add_filter('_wp_post_revision_fields', array($this, 'post_revision_fields'), 10, 1);
-					add_filter('_wp_post_revision_field_postmeta', array($this, 'post_revision_field'), 1, 2);
-				}
-			}
-		}
-	
-		public function register($postmeta_key, $display_func = '') {
-			if (!in_array($postmeta_key, $this->postmeta_keys, true)) {
-				$this->postmeta_keys[] = compact('postmeta_key', 'display_func');
-			}
-			return true;
+/**
+ * Adding admin menu for Revision meta settings.
+ *
+ * @since 2.0.0
+ */
+function cfrm_wp_revisions_manager() {
+	add_submenu_page(
+		'options-general.php',
+		'Revisions Manager Settings',
+		'Revision Manager',
+		'administrator',
+		'wp_revisions_manager',
+		'cfrm_revisions_manager_list'
+		);
+
+}
+add_action('admin_menu', 'cfrm_wp_revisions_manager');
+
+/**
+ * Callback function for revisions settings page.
+ *
+ * @since 2.0.0
+ */
+function cfrm_revisions_manager_list() {
+	//Handles Form Submission.
+	cfrm_revision_process_submission();
+	//Display list of all metas.
+	cfrm_revision_display_metas();
+}
+
+/**
+ * Processes form revision manager form submission.
+ *
+ * @since 2.0.0
+ */
+function cfrm_revision_process_submission() {
+	if ( isset( $_POST['cfrm_exclude_submit'] ) ) {
+		if ( isset( $_POST['cfrm_exclude_meta'] ) ) {
+			$exclude_metas = $_POST['cfrm_exclude_meta'];
+			update_option( 'cfrm_exclude_meta', $exclude_metas );
 		}
-	
-		/**
-		 * This is a paranoid check. There will be no object to register the 
-		 * actions and filters if nobody adds any postmeta to be handled
-		 *
-		 * @return bool
-		 */
-		public function have_keys() {
-			return (bool) count($this->postmeta_keys);
+	}
+}
+
+/**
+ * Outputs html for the form meta list.
+ *
+ * @since 2.0.0
+ */
+function cfrm_revision_display_metas() {
+	$meta_keys = cfrm_get_meta_keys();
+
+	$html = cfrm_get_page_header();
+	$html .= cfrm_get_page_meta_list( $meta_keys );
+	$html .= cfrm_get_page_footer();
+
+	echo $html;
+}
+
+/**
+ * Returns a list of all metas associated with any post type.
+ *
+ * @since  2.0.0
+ *
+ * @return Array An array of all meta keys.
+ */
+function cfrm_get_meta_keys() {
+	$postmeta_keys = array();
+	$posts = get_posts( array( 'post_type' => get_post_types(), 'posts_per_page' => -1 ) );
+	foreach ( $posts as $post ) {
+		$meta_keys = cfrm_get_post_metas( $post->ID );
+		$postmeta_keys = array_unique( array_merge( $postmeta_keys, $meta_keys ) );
+	}
+	return array_values( $postmeta_keys );
+}
+
+/**
+ * Get List of meta keys associated with a particular post.
+ *
+ * @since  2.0.0
+ *
+ * @param  $post_id Integer Post ID.
+ * @return          Array   An array of all metakeys associated with the post.
+ */
+function cfrm_get_post_metas( $post_id ) {
```

Comment:
Since we're only getting the keys, that should be part of the name: `cfrm_get_post_meta_keys( $post_id = 0 )`